models/balancer/arbitrum/balancer_v2_arbitrum_liquidity.sql

```diff
@@ -165,41 +175,46 @@ zipped_balance_changes AS (
             symbol AS token_symbol,
             cumulative_amount as token_balance_raw,
             cumulative_amount / POWER(10, COALESCE(t.decimals, p1.decimals)) AS token_balance,
-            cumulative_amount / POWER(10, COALESCE(t.decimals, p1.decimals)) * COALESCE(p1.price, p2.price, 0) AS amount_usd
+            cumulative_amount / POWER(10, COALESCE(t.decimals, p1.decimals)) * COALESCE(p1.price, p2.price, 0) AS protocol_liquidity_usd,
+            cumulative_amount / POWER(10, COALESCE(t.decimals, p1.decimals)) * COALESCE(p1.price, p2.price, p3.bpt_price) AS pool_liquidity_usd
         FROM calendar c
         LEFT JOIN cumulative_balance b ON b.day <= c.day
         AND c.day < b.day_of_next_change
-        LEFT JOIN {{ ref('tokens_erc20_legacy') }} t ON t.contract_address = b.token
+        LEFT JOIN {{ ref('tokens_erc20') }} t ON t.contract_address = b.token
         AND blockchain = "arbitrum"
         LEFT JOIN prices p1 ON p1.day = b.day
         AND p1.token = b.token
         LEFT JOIN dex_prices p2 ON p2.day <= c.day
         AND c.day < p2.day_of_next_change
         AND p2.token = b.token
+        LEFT JOIN bpt_prices p3 ON p3.day = b.day AND p3.token = CAST(b.token as varchar(42))
```

Comment:
not urgent, but we have to agree on the types of addresses to avoid having the need for casting everywhere